client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -301,6 +301,50 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerI
     }
 
     @Override
+    public CompletableFuture<Boolean> truncateStream(final String scope, final String stream, final StreamCut streamCut) {
+        return truncateStream(scope, stream, streamCut.getPositions().entrySet()
+                .stream().collect(Collectors.toMap(x -> x.getKey().getSegmentNumber(), Map.Entry::getValue)));
+    }
+
+    @Override
+    public CompletableFuture<Boolean> truncateStream(final String scope, final String stream, final Map<Integer, Long> streamCut) {
```

Comment:
Is this call idempotent? What happens if I call truncate stream twice with the same stream cut?

Reply:
So it depends on when you call it:
1. if previous processing is going on and it is called, it will fail.
2. if previous processing has completed, then since this is a valid input (gt eq to previous stream cut) so processing will be done, but the processing is idempotent.

Reply:
Ok.